src/Umbraco.Core/Configuration/Models/ContentSettings.cs

```diff
@@ -16,11 +15,25 @@ public class ContentSettings
 
         public bool ResolveUrlsFromTextString { get; set; } = false;
 
-        public IEnumerable<ContentErrorPage> Error404Collection { get; set; } = Array.Empty<ContentErrorPage>();
+        public ContentErrorPage[] Error404Collection { get; set; } = Array.Empty<ContentErrorPage>();
 
         public string PreviewBadge { get; set; } = DefaultPreviewBadge;
 
-        public MacroErrorBehaviour MacroErrors { get; set; } = MacroErrorBehaviour.Inline;
+        // We could bind the enum MacroErrorsBehaviour directly from configuration, but we're doing so
+        // via this string to allow for validation to occur on start-up that the configured value does
+        // match one of the enum values.
+        // Without this it'll fail on first use, and we'd have less control over the error message.
+        internal string MacroErrors { get; set; } = MacroErrorBehaviour.Inline.ToString();
+
+        public MacroErrorBehaviour MacroErrorsBehaviour
+        {
+            get
+            {
+                return Enum.TryParse<MacroErrorBehaviour>(MacroErrors, true, out var value)
```

Comment:
I'm not sold on this. I would want the framework to throw on invalid configuration not set a value for me.

Reply:
Thanks for taking a look over @JimBobSquarePants.

In practice, the answer to _What happens if I explicitly chose different behavior but spelt it incorrectly?_ would be that this would be caught by the validation I've added in this PR (see `ContentSettingsValidator`).  That's going to make sure that if you've provided a value, it must be one of the accepted enum values, hence you should never get the "assumed" value set here.

But that said, it follows - and I take your point - that when this code is run that validation should have occurred, and so if `TryParse` failed here, it would be an exception (i.e. something has gone wrong if an invalid value hasn't already been caught by the validation), so it's best to throw here rather than silently using a default value.  Have updated.

Reply:
Doesn't that now mean double validation?

You've got code now for a scenario that should never present itself and that a unit test should prevent someone ever changing the code to avoid the default validation.

Reply:
Unfortunately I've had to revert this - it seems that this property is getting evaluated at start-up before the validation occurs, so an invalid entry in configuration will cause this to throw an exception before we get to the validation.

Arguably we could just not validate enum settings and rely on this built-in behaviour if we try to bind directly from configuration to an enum value on our configuration POCO.  Which would also mean we could get rid of the extra string field.  Only reason not to is to be able to handle the validation all in the same way, with similar error messages etc. - which I quite like, but perhaps it's not adding that much value.

Reply:
I would say go with the default framework behavior for configuration driven scenarios. 

For code driven configuration I would normally handle property validation within the setter of the property itself without the need for an additional method to call from somewhere else.

Reply:
OK - will do this (update coming later or tomorrow).  Possibly I'm getting a little too attached to the custom code... but it's probably not adding that much value, and the workarounds introduced are a little confusing.

Reply:
Updated now to that effect.